src/app/auth/auth.module.ts

```diff
@@ -0,0 +1,25 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RegisterComponent } from './register/register.component';
+import { LoginComponent } from './login/login.component';
+import {AuthService} from "./service/auth.service";
+import {HttpClientModule} from "@angular/common/http";
+import {AppRoutingModule} from "../app-routing.module";
+import {ReactiveFormsModule} from "@angular/forms";
+
+@NgModule({
+  declarations: [
+    RegisterComponent,
+    LoginComponent
+  ],
+  imports: [
+    CommonModule,
+    HttpClientModule,
+    AppRoutingModule,
+    ReactiveFormsModule
+  ],
+  providers:[
+    AuthService
```

Comment:
If I'm right, you don't have to specify it here, since you've already injected it into root
```
@Injectable({
  providedIn: 'root'
})
```